.github/workflows/go-test.yml

```diff
@@ -0,0 +1,26 @@
+name: Go tests
+on: [push]
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - windows-latest
+          - ubuntu-latest
+        go-version:
+          - 1.18.x
+          - 1.22.x
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Configure git in Windows
+        run: |
+          git config --global core.autocrlf input
+        if: ${{ matrix.os == 'windows-latest' }}
+      - uses: actions/checkout@v4
+      - name: Setup Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Test with the Go CLI
+        run: go test -v ./...
```

Comment:
Nice ! 

Could it be added also the lib to run static check (`honnef.co/go/tools/cmd/staticcheck`) and goimports? As it is run in elastic-package or other repositories. It could be done in a follow-up PR too.

I've tried locally and it reports some errors/warnings.